### ISSUE TITLE
Suggest modern `[[` instead of `[` or `test` in Bash

### ISSUE THREAD

**Issue by l0b0** [koalaman/shellcheck#2590]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Update: https://www.shellcheck.net/wiki/SC2292 is similar (Does it apply to `test` as well? I'd assume so…), but is not enabled at https://www.shellcheck.net/. Do we need an "all rules enabled" version of that page linked from the issue template?

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
[ -n "$foo" ]
test -n "$foo"
```

#### Here's what shellcheck currently says:

> No issues detected!

#### Here's what I wanted or expected to see:

> Use "[[" instead of "["
> Use "[[" instead of "test"

[Rationale](https://paperless.blog/bash-lints-beyond-shellcheck): `[[` is generally safer and saner than `[`/`test`. See for example [`[[` Is a Builtin, But `[[` Is Part of the Language](http://www.oilshell.org/blog/2016/10/12.html) and [What is the difference between test, `[` and `[[`?](http://mywiki.wooledge.org/BashFAQ/031).

**Comment by brother** [comment]
As mentioned in the linked wiki page (SC2292) this is an [optional](https://github.com/koalaman/shellcheck/wiki/Optional) check and must be activated explicit, shellcheck.net is operating in a "default mode".
Suggestions for changes to shellcheck.net is better discussed at that [issue board](https://github.com/koalaman/shellcheck.net/issues) - for instance adding a checkbox or something for "enable all optinal rules" or something.

`shellcheck -o require-double-brackets myscript.sh`

```
name:    require-double-brackets
desc:    Require [[ and warn about [ in Bash/Ksh
example: [ -e /etc/issue ]
fix:     [[ -e /etc/issue ]]
```
